sdlib/d/gc/page.d

```diff
@@ -726,9 +752,18 @@ private:
 					auto newOccupancy = oldOccupancy & bmp[i];
 
 					occupancyMask |= newOccupancy;
-					count += popCount(oldOccupancy ^ newOccupancy);
+					auto toRemove = oldOccupancy ^ newOccupancy;
+					count += popCount(toRemove);
 
-					e.slabData.rawContent[i] = newOccupancy;
+					scope(success) e.slabData.rawContent[i] = newOccupancy;
+
+					if (!ec.supportsMetadata) {
+						continue;
+					}
+
+					finalizeSlabNimble(
+						toRemove & e.slabMetadataFlags.rawContent[i], sc, e,
```

Comment:
`toRemove & e.slabMetadataFlags.rawContent[i]` can be extracted out of the call.